service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseManagementOrderService.java

```diff
@@ -39,7 +77,47 @@ public CaseManagementOrder addAction(CaseManagementOrder order, OrderAction orde
         return data;
     }
 
+    public CaseManagementOrder addHearingDetailsToCMO(DynamicList list, CaseManagementOrder order) {
+        CaseManagementOrder.CaseManagementOrderBuilder builder = order.toBuilder();
```

Comment:
Looks like we're creating unnecessary instances of `CaseManagementOrder` and `CaseManagementOrder.CaseManagementOrderBuilder` every time `list` is null. Can we simply:
```
if (list == null) {
  return order;
}
```